config/example/node.yaml

```diff
@@ -1,159 +1,159 @@
 
-mainchain:
-  dial_timeout: 30s
-  rpc_endpoint:
+mainchain:  # DEPRECATED section, is not used and not read
+  dial_timeout: 30s  # timeout for main chain NEO RPC client connection
+  rpc_endpoint:  # main chain NEO RPC endpoints; are shuffled and used one by one until the first success
     - https://rpc1.n3.nspcc.ru:30341
     - https://rpc2.n3.nspcc.ru:30341
 
 apiclient:
-  dial_timeout: 15s
+  dial_timeout: 15s  # timeout for NEOFS API client connection
 
 policer:
-  head_timeout: 15s
+  head_timeout: 15s  # timeout for the Policer HEAD remote operation
 
 replicator:
-  put_timeout: 15s
+  put_timeout: 15s  # timeout for the Replicator PUT remote operation
 
 object:
   put:
-    pool_size_remote: 100
-    pool_size_local: 101
+    pool_size_remote: 100  # number of async workers for remote PUT operations
+    pool_size_local: 101  # number of async workers for local PUT operations
 
 storage:
-  shard_num: 2
+  shard_num: 2  # total number of shards
   shard:
     0:
-      refill_metabase: false
+      refill_metabase: false  # sync metabase with blobstore on start, expensive, leave false until complete understanding
 
-      use_write_cache: false
+      use_write_cache: false  # use write-cache
 
       writecache:
-        path: tmp/0/cache
-        mem_size: 2147483648
-        small_size: 16384
-        max_size: 134217728
-        workers_number: 30
-        size_limit: 3221225472
+        path: tmp/0/cache  # write-cache root directory
+        mem_size: 2147483648  # approximate RAM usage limit for "small" objects, bytes
+        small_size: 16384  # size threshold for "small" objects which are cached in key-value DB, not on FS, bytes
+        max_size: 134217728  # size threshold for "big" objects which bypass write-cache and go to the storage directly, bytes
+        workers_number: 30  # number of write-cache flusher threads
+        size_limit: 3221225472  # approximate write-cache total size, bytes
 
       metabase:
-        path: tmp/0/meta
-        perm: 0644
+        path: tmp/0/meta  # metabase path
+        perm: 0644  # permissions for metabase files(directories: +x for current user and group)
 
       blobstor:
-        path: tmp/0/blob
-        perm: 0644
-        compress: true
-        shallow_depth: 5
-        small_size_limit: 102400
+        path: tmp/0/blob  # blobstor path
+        perm: 0644  # permissions for blobstor files(directories: +x for current user and group)
+        compress: true  # turn on/off zstd(level 3) compression of stored objects
+        shallow_depth: 5  # directory structure depth
+        small_size_limit: 102400  # size threshold for objects that are considered to be "small", bytes
 
         blobovnicza:
-          size: 4194304
-          shallow_depth: 1
-          shallow_width: 4
-          opened_cache_size: 50
+          size: 4194304  # approximate size limit of single blobovnicza instance, total size will be: size*width^(depth+1), bytes
```

Comment:
As a random user, I have no idea what size of blobovnicza with all its dimensions should I have on my machine. Is `shallow_depth: 1; shallow_width: 4` and `shallow_depth: 0; shallow_width: 8` the same or not?

I wonder if this section ever should be configurable by a user. Maybe, we should not expose that much internal parameters. Instead, we might somehow provide pre-configurable blobovniczy with description for every configuration and when it is most applicable. What do you think?